src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -50,11 +51,25 @@ public Publish(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFac
         public async Task<IEnumerable<PublishResult>> ProcessAsync(string folderPath)
         {
             this.publishResults.Clear();
+
             if (!this.IsValidInput(folderPath))
             {
                 return this.publishResults;
             }
 
+            var productValidate = new ValidateProduct();
+            var errors = productValidate.IsProductValid(folderPath);
+            foreach (var error in errors)
+            {
+                this.publishResults.Add(error);
+            }
+
+            if (errors.Any(r => r.IsError))
+            {
+                return this.publishResults;
+            }
+
+            this.publishResults.Add(new PublishResult { IsError = false, ResultCode = ResultCode.ProductValidated, Message = "Product documents validated" });
             using (var publishYml = File.OpenText(Path.Combine(folderPath, @"publish.yml")))
```

Comment:
Change `using` to simple form https://docs.microsoft.com/pl-pl/dotnet/csharp/whats-new/csharp-8#using-declarations

Reply:
Done